.github/workflows/artifacts-build.yml

```diff
@@ -50,6 +50,7 @@ jobs:
     env:
       GCP_PROJECT_ID: "dev-infra-273822"
       GCP_BUCKET_ID: "icu4x-pr-artifacts"
+      RUSTDOCFLAGS: "-Dwarnings"
```

Comment:
Issue: This is not the right place to put this warning.  This job runs on push and within the context of the repository.  If this fails, there will be no indication on the pull request.

I suggest adding this to the main build-test.yml file somewhere.

Reply:
I don't think build-test.yml runs rustdoc? I was under the impression that warnings from this task show up on the PR as the fake "comments" that GHA posts (like it does for clippy) but on further looking that's definitely not the case.

Reply:
Perhaps the check job should also be running doc.

Reply:
Does `cargo doc` imply `cargo check`?  Is there an efficiently place to put it or does it need to go into the big build job?

Reply:
`cargo doc` _kinda_ runs `cargo check` but not quite, and I would not rely on it for check diagnostics.

If you run `cargo check` first then `cargo doc` ought to run rather quickly. I would suggest running check then doc in the check job.

Reply:
Seems reasonable.  So let's put this into `ci-job-check`.

We should also *undo* the change in this PR, because I think it's counterproductive to halt the CI artifacts job.  I would rather have the docs uploaded to CI despite warnings instead of blocked for an error that no one will ever see.

Reply:
yeah, makes sense